File: hyeonsook95/baekjoon/1913.py
import sys

input = sys.stdin.readline

N = int(input().strip())
maps = [[0 for _ in range(N)] for _ in range(N)]

dr, dc = [1, 0, -1, 0], [0, 1, 0, -1]

d_idx = 0
r_idx, c_idx = 0, 0
for num in range(N**2, 0, -1):
    maps[r_idx][c_idx] = num
    if not (
        -1 < r_idx + dr[d_idx] < N
        and -1 < c_idx + dc[d_idx] < N
        and maps[r_idx + dr[d_idx]][c_idx + dc[d_idx]] == 0
    ):
        d_idx = (d_idx + 1) % 4
    r_idx += dr[d_idx]
    c_idx += dc[d_idx]

for m in maps:
    print(*m, sep=" ", end="\n")

File: hyeonsook95/baekjoon/1913.ts
var fs = require("fs");
var input = fs.readFileSync('dev/stdin').toString().split('\n');

const N = parseInt(input[0]);
const target = parseInt(input[1]);

var tr = -1;
var tc = -1;
var maps = Array.from(Array(N), () => Array(N).fill(0));

var didx = 0;
const dr = [1, 0, -1, 0];
const dc = [0, 1, 0, -1];

var ridx = 0;
var cidx = 0;

for (var num = N ** 2; num > 0; num--) {
    if (num === target) {
        tr = ridx + 1;
        tc = cidx + 1;
    }

    maps[ridx][cidx] = num;
    if ((0 > ridx + dr[didx]) || (N - 1 < ridx + dr[didx]) || (0 > cidx + dc[didx]) || (N - 1 < cidx + dc[didx]) || (maps[ridx + dr[didx]][cidx + dc[didx]] !== 0)) {
        didx = (didx + 1) % 4;
    }
    ridx = ridx + dr[didx];
    cidx = cidx + dc[didx];

}

for (var row of maps) {
    console.log(...row);
}
console.log(tr, tc);